### PR TITLE
Fix #247: autosummary: Add autosummary_generate_option to overwrite old stub file

### LINKED ISSUE #247: autosummary: generated stub files are not re-written
If the top-level autosummary directive is changed from one sphinx run to the other (e.g. a new option added or the template was changed) the generated stub files in the directory given with the :toctree: option are not automatically updated.

The are onyl re-created if the stub files direcory is deleted manually from the file explorer.

---
- Bitbucket: https://bitbucket.org/birkenfeld/sphinx/issue/247
- Originally reported by: [timmie](https://bitbucket.org/timmie)
- Originally created at: 2009-08-29T17:49:55.802

### PR DESCRIPTION
**Pull request by tk0miya**
### Feature or Bugfix
- Feature

### Purpose
- refs: #247 
- Add `autosummary_generate_option` to skip, warn or overwrite for existing stub files.
- IMO, it would be better to use `warning` by default. But it introduces a little breaking change. So this only adds config variables. And I will change the default settings to `warning` in 3.0.

### CODE REVIEW

File: sphinx/ext/autosummary/__init__.py [hunk 1]
```diff
@@ -780,6 +782,8 @@ def setup(app):
     app.connect('doctree-read', process_autosummary_toc)
     app.connect('builder-inited', process_generate_options)
     app.add_config_value('autosummary_generate', [], True, [bool])
+    app.add_config_value('autosummary_generate_option', 'skip', False,
+                         ENUM('skip', 'warning', 'overwrite'))
```
Comment (tk0miya) [inline sphinx/ext/autosummary/__init__.py]: I need the better name for this configuration...

Comment (shimizukawa) [reply sphinx/ext/autosummary/__init__.py]: I want to avoid autosummary_generate_option, because the term option associates the option of the directive.

my suggestions:

* autosummary_generate_behavior
* autosummary_generate_confirmation (too long ..)

Comment (tk0miya) [reply sphinx/ext/autosummary/__init__.py]: Thank you for comment. I reconsider this option again.

The `skip` will not help users. Because it hides conflicts. That causes the outputs are not updated. This is only added to keep compatibility. It is meaningless to me.

Next, the `warning` is a bit better than `skip`. It notices conflicts to users. Users can clear them by removing old stub files. I think the warning only helps who put user content as same name as stub file. Almost people feel harmful because they need to remove old stub files manually. So I think this option is useful only at first run.

The `overwrite` is a best way for almost case. Almost users will choose this option finally, I believe.

In conclusion, I think `autosummary_generate_overwrite = True | False` would be better than `autosummary_generate_option`. I withdraw `skip` option, and will be merged into 3.0 release because of breaking change.

What do you think?

Comment (shimizukawa) [reply sphinx/ext/autosummary/__init__.py]: +1 for `autosummary_generate_overwrite`. 
Thanks for your summarizing!

**Review by shimizukawa — APPROVED** [review]
+1 for `autosummary_generate_overwrite`.
LGTM!
